### PR TITLE
BERT with torch.compile

### PR DESCRIPTION
**Pull request by agunapal**
## Description

This updates the BERT example to also use torch.compile

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
Logs for Test A

- [ ] Test B
Logs for Test B

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: examples/Huggingface_Transformers/README.md [hunk 1]
```diff
@@ -18,22 +18,37 @@ To get started [install Torchserve](https://github.com/pytorch/serve) and then
...
+    mode: reduce-overhead
+```
+When batch_size is 1, for BERT models, the operations are memory bound. Hence, we make use of `reduce-overhead` mode to make use of CUDAGraph and get better performance.
+
+To use PyTorch Eager or TorchScript, you can remove the above config.
+
 For Torchscript support, check out [torchscript.md](torchscript.md)
 
-#### Setting the setup_config.json
+#### Setting the  handler config in model-config.yaml
```
Comment (mreso) [inline examples/Huggingface_Transformers/README.md:49]:
```suggestion
#### Setting the handler config in model-config.yaml
```

File: examples/Huggingface_Transformers/Transformer_handler_generalized.py [hunk 2]
```diff
@@ -40,6 +40,8 @@ def initialize(self, ctx):
             pertaining to the model artifacts parameters.
         """
         self.manifest = ctx.manifest
+        if ctx is not None and hasattr(ctx, "model_yaml_config"):
+            self.model_yaml_config = ctx.model_yaml_config
```
Comment (mreso) [inline examples/Huggingface_Transformers/Transformer_handler_generalized.py:44]: Will be uninitialized with not in ctx. Better to initialize with {}

File: examples/Huggingface_Transformers/Transformer_handler_generalized.py [hunk 3]
```diff
@@ -50,20 +52,13 @@ def initialize(self, ctx):
...
         )
-        # read configs for the mode, model_name, etc. from setup_config.json
-        setup_config_path = os.path.join(model_dir, "setup_config.json")
-        if os.path.isfile(setup_config_path):
-            with open(setup_config_path) as setup_config_file:
-                self.setup_config = json.load(setup_config_file)
+
+        # read configs for the mode, model_name, etc. from the handler config
+        if hasattr(self, "model_yaml_config") and "handler" in self.model_yaml_config:
+            self.setup_config = self.model_yaml_config["handler"]
```
Comment (mreso) [inline examples/Huggingface_Transformers/Transformer_handler_generalized.py:58]: See above, then use:
```suggestion
        self.setup_config = self.model_yaml_config.get("handler", None)
```

File: examples/Huggingface_Transformers/Transformer_handler_generalized.py [hunk 4]
```diff
@@ -50,20 +52,13 @@ def initialize(self, ctx):
...
             else "cpu"
         )
-        # read configs for the mode, model_name, etc. from setup_config.json
-        setup_config_path = os.path.join(model_dir, "setup_config.json")
-        if os.path.isfile(setup_config_path):
-            with open(setup_config_path) as setup_config_file:
-                self.setup_config = json.load(setup_config_file)
+
+        # read configs for the mode, model_name, etc. from the handler config
+        if hasattr(self, "model_yaml_config") and "handler" in self.model_yaml_config:
```
Comment (mreso) [inline examples/Huggingface_Transformers/Transformer_handler_generalized.py:57]:
```suggestion
```

File: examples/Huggingface_Transformers/Transformer_handler_generalized.py [hunk 5]
```diff
@@ -50,20 +52,13 @@ def initialize(self, ctx):
...
+            self.setup_config = self.model_yaml_config["handler"]
         else:
```
Comment (mreso) [inline examples/Huggingface_Transformers/Transformer_handler_generalized.py:59]:
```suggestion
        if not self.setup_config:
```

File: examples/Huggingface_Transformers/Transformer_handler_generalized.py [hunk 6]
```diff
@@ -228,7 +240,8 @@ def inference(self, input_batch):
         inferences = []
         # Handling inference for sequence_classification.
         if self.setup_config["mode"] == "sequence_classification":
-            predictions = self.model(input_ids_batch, attention_mask_batch)
+            with torch.inference_mode():
```
Comment (mreso) [inline examples/Huggingface_Transformers/Transformer_handler_generalized.py:243]: Why not use `@torch.inference_mode()` decorator on `def inference()`

**Review by mreso — APPROVED** [review]
Left some minor comments. LGTM otherwise

File: examples/Huggingface_Transformers/Transformer_handler_generalized.py (hunk 2 above)
Comment (agunapal) [reply examples/Huggingface_Transformers/Transformer_handler_generalized.py]: done

File: examples/Huggingface_Transformers/Transformer_handler_generalized.py (hunk 3 above)
Comment (agunapal) [reply examples/Huggingface_Transformers/Transformer_handler_generalized.py]: done

File: examples/Huggingface_Transformers/Transformer_handler_generalized.py (hunk 5 above)
Comment (agunapal) [reply examples/Huggingface_Transformers/Transformer_handler_generalized.py]: done

File: examples/Huggingface_Transformers/Transformer_handler_generalized.py (hunk 6 above)
Comment (agunapal) [reply examples/Huggingface_Transformers/Transformer_handler_generalized.py]: done

**Pull request merged**